### PR TITLE
1-5-18 testing

### PR DESCRIPTION
**Pull request by thehabes**
Put commits to this branch or comment on the pull request with bugs you found @cubap.

### CODE REVIEW

File: src/java/edu/slu/action/ObjectAction.java [hunk 1]
```diff
@@ -740,6 +768,16 @@ public void saveNewObject() throws IOException, ServletException, Exception{
             //set @id from _id and update the annotation
             BasicDBObject dboWithObjectID = new BasicDBObject((BasicDBObject)dbo);
             String uid = "http://devstore.rerum.io/rerumserver/id/"+newObjectID;
+            if(dboWithObjectID.containsKey("@id")){
+                //Someone tried to create an object that already had an @id.  How do we want to handle this?
```
Comment (cubap) [inline src/java/edu/slu/action/ObjectAction.java]: I thought we were rejecting create actions with @id. I think we need to throw a method error with a note to Update objects with @ids.

File: src/java/edu/slu/action/ObjectAction.java [hunk 2]
```diff
@@ -727,6 +754,7 @@ public void getByProperties() throws IOException, ServletException, Exception{
      */
     public void saveNewObject() throws IOException, ServletException, Exception{
         if(null != processRequestBody(request, false) && methodApproval(request, "create")){
+            boolean intendedToHaveNewID = true;
```
Comment (cubap) [inline src/java/edu/slu/action/ObjectAction.java]: see below, but this flag doesn't make sense. If a user wants to add it, they just remove the property.

File: src/java/edu/slu/action/ObjectAction.java [hunk 3]
```diff
@@ -793,73 +831,82 @@ public void updateObject() throws IOException, ServletException, Exception{
...
+                            Logger.getLogger(ObjectAction.class.getName()).log(Level.SEVERE, null, ex);
+                        }
+                    }
+                    else{
+                        //The error is already written to response.out, do nothing.
                     }
                 }
                 else{
-                    //The error is already written to response.out, do nothing.
+                    writeErrorResponse("Object "+received.getString("@id")+" not found in RERUM, could not update.", HttpServletResponse.SC_BAD_REQUEST);
```
Comment (cubap) [inline src/java/edu/slu/action/ObjectAction.java]: This case is actually the remote object case, isn't it? Shouldn't we create and add the @id to __rerum.history.previous?

File: src/java/edu/slu/action/ObjectAction.java (hunk 2 above)
Comment (thehabes) [reply src/java/edu/slu/action/ObjectAction.java]: agreed.

File: src/java/edu/slu/action/ObjectAction.java (hunk 1 above)
Comment (thehabes) [reply src/java/edu/slu/action/ObjectAction.java]: done deal

File: src/java/edu/slu/action/ObjectAction.java (hunk 3 above)
Comment (thehabes) [reply src/java/edu/slu/action/ObjectAction.java]: We need to break this out into its own ticket.  We need to rethink what update.action does and how to separate and handle PUT vs PATCH gracefully and compliantly.

**Pull request merged**